astropy/table/tests/test_table.py

```diff
@@ -2898,14 +2898,19 @@ def test_table_attribute_ecsv():
 
 
 def test_table_attribute_fail():
-    # Code raises ValueError(f'{attr} not allowed as TableAttribute') but in this
-    # context it gets re-raised as a RuntimeError during class definition.
-    with pytest.raises(RuntimeError, match="Error calling __set_name__"):
+    if sys.version_info[:2] >= (3, 12):
```

Comment:
I'm not sure if `ruff` (pyupgrade) knows how to clean the `else` branch once it becomes dead code.
Specifically, I'm referring to the explicit slicing of `sys.version_info`.
As the author of the original feature in `pyupgrade`, I know that the following pattern *is* auto-fixable when Python 3.12 becomes the minimal supported version
```python
if sys.version_info >= (3, 12):
    ...
else:
    ...
```
but maybe ruff made it a smarter implementation of it ? Anyway, sorry about splitting hairs here.

Reply:
I copied that syntax I have from what I see in the rest of the code base. I am also not sure what is the best way to do this. For dependencies, we have a syntax like `NUMPY_LT_X_Y` that most core maintainers know how to `grep` for when dropping older versions. But I don't think we have one set up for Python, mainly because Python version is basically fixed for a particular environment at run time.

If done manually, this does not matter really because you can still `grep` like this (or something similar):

```
grep "sys\.version" -r . | grep "3, 12"
```